src/main/java/com/jeff_media/updatechecker/ThreadScheduler.java

```diff
@@ -0,0 +1,33 @@
+package com.jeff_media.updatechecker;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+public class ThreadScheduler {
+    public static ScheduledExecutorService runTask(Runnable task) {
+        return runTask(task, 1/20L, TimeUnit.MILLISECONDS);
+    }
+
+    public static ScheduledExecutorService runTask(Runnable task, long delay, TimeUnit timeUnit) {
+        // Create a ScheduledExecutorService with a single thread
+        ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(1);
```

Comment:
The idea is to create only one thread pool and then reuse it every time you want to schedule a task

Reply:
oh okay
I'll edit that